applications/serial_lte_modem/src/gnss/slm_at_gnss.c

```diff
@@ -122,15 +150,206 @@ static void pgps_req_wk(struct k_work *work)
 	}
 }
 
+AT_MONITOR(ncell_meas, "NCELLMEAS", ncell_meas_mon, PAUSED);
+
+static void ncell_meas_mon(const char *notify)
+{
+	int err;
+	uint32_t param_count;
+	char tmp[8] = {0};
+
+	ncell_meas_status = -1;
+	at_params_list_clear(&at_param_list);
+	err = at_parser_params_from_str(notify, NULL, &at_param_list);
+	if (err) {
+		goto exit;
+	}
+
+	/* parse status, 0: success 1: fail */
+	err = at_params_int_get(&at_param_list, 1, &ncell_meas_status);
+	if (err) {
+		goto exit;
+	}
+	if (ncell_meas_status != 0) {
+		LOG_ERR("NCELLMEAS failed");
+		err = -EAGAIN;
+		goto exit;
+	}
+	param_count = at_params_valid_count_get(&at_param_list);
+	if (param_count < MAX_PARAM_CELL) { /* at least current cell */
+		LOG_ERR("Missing param in NCELLMEAS notification");
+		err = -EAGAIN;
+		goto exit;
+	}
+
+	/* parse Cell ID */
+	char cid[9];
+	size_t size = sizeof(cid);
+
+	size = sizeof(cid);
+	err = util_string_get(&at_param_list, 2, cid, &size);
+	if (err) {
+		goto exit;
+	}
+	err = util_str_to_int(cid, 16, (int *)&cell_data.current_cell.id);
+	if (err) {
+		goto exit;
+	}
+
+	/* parse PLMN */
+	char plmn[6];
+
+	size = sizeof(plmn);
+	err = util_string_get(&at_param_list, 3, plmn, &size);
+	if (err) {
+		goto exit;
+	}
+	strcpy(tmp, plmn);
+	tmp[3] = '\0'; /* MCC always 3-digit */
+	err = util_str_to_int(tmp, 10, &cell_data.current_cell.mcc);
```

Comment:
Replace tmp with plmn ?

Reply:
We need plmn[] data untouched as next to parse MNC.

Reply:
Ah, I misunderstood, sorry for the noise.

Reply:
Thanks, Boe